src/services/LightStreamer.ts

```diff
@@ -8,63 +8,59 @@ import {
   BlockID,
 } from "@/models/lightstreamer";
 import { ifClient } from "@/utils/ironfish";
+import { lightBlock } from "@/utils/lightBlock";
 import { lightBlockCache } from "@/cache";
-import { lightBlock } from "@/utils/light_block";
+import { ServiceError, handle } from "@/utils/grpc";
 
 class LightStreamer implements LightStreamerServer {
   [method: string]: UntypedHandleCall;
 
-  public getLatestBlock: handleUnaryCall<Empty, BlockID> = async (
-    _,
-    callback,
-  ) => {
+  public getLatestBlock = handle<Empty, BlockID>(async (_, callback) => {
     const rpcClient = await ifClient.getClient();
     const response = await rpcClient.chain.getChainInfo();
     callback(null, {
       sequence: Number(response.content.currentBlockIdentifier.index),
       hash: Buffer.from(response.content.currentBlockIdentifier.hash, "hex"),
     });
-  };
+  });
 
-  public getBlock: handleUnaryCall<BlockID, LightBlock> = async (
-    call,
-    callback,
-  ) => {
+  public getBlock = handle<BlockID, LightBlock>(async (call, callback) => {
     if (!call.request.hash && !call.request.sequence) {
-      callback(new Error("Must provide either hash or sequence"), null);
+      throw new ServiceError(
+        status.INVALID_ARGUMENT,
+        "Either hash or sequence must be provided",
+      );
```

Comment:
The `handle` function lets you throw errors instead having to `callback(err, null)`.